cmd/kubeadm/app/util/kubeconfig_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+)
+
+type configClient struct {
+	c  string
+	s  string
+	ca []byte
+}
+
+type configClientWithCerts struct {
+	c           *clientcmdapi.Config
+	clusterName string
+	userName    string
+	clientKey   []byte
+	clientCert  []byte
+}
+
+type configClientWithToken struct {
+	c           *clientcmdapi.Config
+	clusterName string
+	userName    string
+	token       string
+}
+
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+func TestCreateBasicClientConfig(t *testing.T) {
+	var createBasicTest = []struct {
+		cc       configClient
+		expected string
+	}{
+		{configClient{}, ""},
+		{configClient{c: "kubernetes"}, ""},
+	}
+	for _, rt := range createBasicTest {
+		c := CreateBasicClientConfig(rt.cc.c, rt.cc.s, rt.cc.ca)
+		if c.Kind != rt.expected {
```

Comment:
This seems like a pretty weak test.  Is this just testing that things don't blow up?